scripts/__utils__.sh

```diff
@@ -18,7 +24,14 @@ function log() {
 }
 
 function hmac_sha256() {
-   printf "$2" | openssl dgst -binary -hex -sha256 -mac HMAC -macopt hexkey:"$1"
+  uname | grep -q ^Darwin
+  if [ $? -eq 0 ]; then
+    local hash="$(printf "$2" | ${OPENSSL_BIN} dgst -binary -sha256 -mac HMAC -macopt hexkey:"$1" | xxd -p -c 256)"
+  else
+    local hash="$(printf "$2" | openssl dgst -binary -hex -sha256 -mac HMAC -macopt hexkey:"$1")"
+  fi
+
+  echo "${hash}"
 }
 
 __utils__=true
```

Comment:
we could reformat this as discussed at https://github.com/riboseinc/aws-authenticating-secgroup-scripts/pull/6#issuecomment-355720664